### DISCUSSION TITLE
Symfony Lock does not work as I expect

### DISCUSSION THREAD

**Discussion by bartholdbos** [symfony/symfony#44298] [Q&A]
I want to use the Symfony Lock component to prevent a race condition with the UniqueEntity Constraint. In my controller I use Validation to validate that the entity is unique after which it is persisted to the database. However when a user clicks the link twice both requests succeed in persisting to the database. I have injected the lock factory and create a lock in the beginning of the action function

I have made a small code sample to demonstrate my issue. When making the second request from a different browser or incognito window the second request fails to acquire the lock. But if two requests are made from the same browser both can acquire the lock. I am using Symfony 5.3 and PHP 7.4 with apache mod_php

```
<?php

namespace App\Controller;

use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\Lock\LockFactory;
use Symfony\Component\Routing\Annotation\Route;

class LockTest extends AbstractController
{
    /**
     * @Route("/test")
     * @Template("lock/test.html.twig")
     */
    public function test(LockFactory $factory): array
    {
        $lock = $factory->createLock("test");

        $acquired = $lock->acquire();

        dump($lock, $acquired);

        sleep(2);

        dump($lock->isAcquired());

        return ["message" => "testing"];
    }
}
```

**Comment by HugoDiag** [comment]
You should try using `$lock->acquire(true)`.
By default, acquire is non blocking

**Comment by mwoelke** [comment]
Did you ever fix this?

I am seeing the same problem where the same browser can acquire the lock multiple times while other browsers fail. Using the SemaphoreLock.
symfony/lock version 5.4.40

For now I will switch to RedisLock hoping this will fix the problem.
